app/full_text_search.py

```diff
@@ -143,19 +145,20 @@ def create_jp_name_fields(**kwargs):
 def create_jp_location_fields(**kwargs):
```

Comment:
Now you support multiple readings, so how about combining name dictionary and location dictionary, and applying it for both names and locations? You can add TODO comment and do it later.